packages/dataviews/src/dataviews-layouts/table/index.tsx

```diff
@@ -102,13 +110,32 @@ function TableColumnField< Item >( {
 	primaryField,
 	item,
 	field,
+	isClickable,
+	onClick,
 }: TableColumnFieldProps< Item > ) {
 	return (
 		<div
 			className={ clsx( 'dataviews-view-table__cell-content-wrapper', {
 				'dataviews-view-table__primary-field':
 					primaryField?.id === field.id,
+				'dataviews-view-table__primary-field--clickable':
+					isClickable( item ),
 			} ) }
+			tabIndex={ isClickable( item ) ? 0 : undefined }
+			role="button"
+			onClick={ () => {
+				if ( isClickable( item ) ) {
+					onClick( item );
+				}
+			} }
+			onKeyDown={ ( event ) => {
+				if (
+					( event.key === 'Enter' || event.key === '' ) &&
+					isClickable( item )
+				) {
+					onClick( item );
+				}
+			} }
```

Comment:
This breaks how the `onChangeSelection` works currently and causes `onClick` to be triggered for every column except for the selection column.

Reply:
Could this be only added for the primary field? ( `primaryField?.id === field.id` ? )

Reply:
Thanks for catching that! I fixed the issue:

https://github.com/user-attachments/assets/d34dcfc3-c32c-4003-9860-59c30f94ee15


I wrapped the primary field in a dedicated div to ensure `onItemClick` is invoked only when the user clicks directly on the field.

https://github.com/WordPress/gutenberg/blob/5c7b0c45c8f68f863521f3312c161d8b3a8d9d83/packages/dataviews/src/dataviews-layouts/table/index.tsx#L124-L144